### PR TITLE
Feature/validator register api

### PR DESCRIPTION
**Pull request by talekhinezh**

### CODE REVIEW

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/application/ValidatorRegistrator.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+	private static final Logger logger = LogManager.getLogger();
+	private final RadixEngine<LedgerAtom> radixEngine;
+	private final RadixAddress self;
+	private final Hasher hasher;
+	private final HashSigner hashSigner;
+	private final Serialization serialization;
+	private final EventDispatcher<MempoolAdd> mempoolAddEventDispatcher;
+
+	// TODO: do something better than this hack
+	@Inject(optional = true)
```
Comment (LukasGasior1) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/application/ValidatorRegistrator.java:65]: in what cases can a fee table be optional?

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/DispatcherModule.java [hunk 2]
```diff
@@ -86,6 +87,8 @@
 
 	@Override
 	public void configure() {
+		bind(new TypeLiteral<EventDispatcher<ValidatorRegistration>>() { })
+				.toProvider(Dispatchers.dispatcherProvider(ValidatorRegistration.class)).in(Scopes.SINGLETON);
```
Comment (LukasGasior1) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/DispatcherModule.java:91]: extra tab

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/application/ValidatorRegistrator.java (hunk 1 above)
Comment (talekhinezh) [reply radixdlt-core/radixdlt/src/main/java/com/radixdlt/application/ValidatorRegistrator.java:65]: testing mostly

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/application/ValidatorState.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+import com.radixdlt.atommodel.validators.RegisteredValidatorParticle;
+
+import java.util.function.Function;
+import java.util.function.LongFunction;
+
+/**
+ * The validator state of a node
+ */
+public final class ValidatorState {
```
[the remark this answers is not in the document]
Comment (talekhinezh) [reply radixdlt-core/radixdlt/src/main/java/com/radixdlt/application/ValidatorState.java:29]: Yes, like a state machine lol. But don't want to overdesign this as it's more of a temporary class.

**Pull request merged**